src/EditorFeatures/CSharpTest/AddParameter/AddParameterTests.cs

```diff
@@ -584,5 +589,1762 @@ void M()
+    {
+        partial void PartialM(int v) { }
+        void M1()
+        {
+            PartialM(1);
+        }
+    }
+}";
+            await TestInRegularAndScriptAsync(code, fix0);
+        }
+
+        [WorkItem(21446, "https://github.com/dotnet/roslyn/issues/21446")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddParameter)]
+        public async Task TestInvocation_Cascading_BaseNotInSource()
+        {
+            // error CS1501: No overload for method 'M' takes 1 arguments
+            var code =
+@"
+<Workspace>
+    <Project Language=""C#"" CommonReferences=""true"">
+        <MetadataReferenceFromSource Language=""C#"" CommonReferences=""true"">
+            <Document FilePath=""ReferencedDocument"">
+namespace N
+{
+    public class BaseClass
+    {
+        public virtual void M() { }
+    }
+}
+            </Document>
+        </MetadataReferenceFromSource>
+        <Document FilePath=""TestDocument"">
+namespace N
+{
+    public class Derived: BaseClass
+    {
+        public void M2()
+        {
+            [|M|](1);
+        }
+    }
+}
+        </Document>
+    </Project>
+</Workspace>";
+            await TestMissingAsync(code);
+        }
+
+        [WorkItem(21446, "https://github.com/dotnet/roslyn/issues/21446")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddParameter)]
+        public async Task TestInvocation_Cascading_RootNotInSource()
+        {
+            // error CS1501: No overload for method 'M' takes 1 arguments
+            var code =
+@"
+<Workspace>
+    <Project Language=""C#"" CommonReferences=""true"">
+        <MetadataReferenceFromSource Language=""C#"" CommonReferences=""true"">
+            <Document FilePath=""ReferencedDocument"">
+namespace N
+{
+    public class BaseClass
+    {
+        public virtual void M() { }
+    }
+}
+            </Document>
+        </MetadataReferenceFromSource>
+        <Document FilePath=""TestDocument"">
+namespace N
+{
+    public class Derived: BaseClass
+    {
+        public override void M() { }
+    }
+    public class DerivedDerived: Derived
+    {
+        public void M2()
+        {
+            [|M|](1);
+        }
+    }
+}
+        </Document>
+    </Project>
+</Workspace>";
+            var fixedDocumentWithoutConflictAnnotation = @"
+namespace N
+{
+    public class Derived: BaseClass
+    {
+        public override void M(int v) { }
+    }
+    public class DerivedDerived: Derived
+    {
+        public void M2()
+        {
+            M(1);
+        }
+    }
+}
+        ";
+            var fixedDocumentWithConflictAnnotation = @"
+namespace N
+{
+    public class Derived: BaseClass
+    {
+        public override void M({|Conflict:int v|}) { }
+    }
+    public class DerivedDerived: Derived
+    {
+        public void M2()
+        {
+            M(1);
+        }
+    }
+}
+        ";
+            await TestInRegularAndScriptAsync(code, fixedDocumentWithoutConflictAnnotation, index: 0);
```

Comment:
I am not sure whether I understand this code. so there is 2 code fix suggested here?

Reply:
One fix is without cascading (only fixing one declaration). The second fix tries to fix all declaration but can't do it for all declarations because some declarations are not in source code (from metadata).

Reply:
tagging @jinujoseph @kuhlenh on declaration only fix.